streams/src/test/java/org/apache/kafka/streams/processor/internals/GlobalStreamThreadTest.java

```diff
@@ -251,6 +264,145 @@ public void shouldDieOnInvalidOffsetException() throws Exception {
             "GlobalStreamThread should have died.");
     }
 
+    @Test
+    public void shouldExpectSkipUnusedTopics() throws Exception {
+        final MockConsumer<byte[], byte[]> tmpMockConsumer1 = new MockConsumer<>(OffsetResetStrategy.NONE);
+        tmpMockConsumer1.updatePartitions(
+            GLOBAL_STORE_TOPIC_NAME,
+            Collections.singletonList(
+                new PartitionInfo(
+                    GLOBAL_STORE_TOPIC_NAME,
+                    0,
+                    null,
+                    new Node[0],
+                    new Node[0])
+            )
+        );
+        tmpMockConsumer1.updateBeginningOffsets(Collections.singletonMap(topicPartition, 0L));
+        tmpMockConsumer1.updateEndOffsets(Collections.singletonMap(topicPartition, 0L));
+        tmpMockConsumer1.assign(Collections.singleton(topicPartition));
+        final MaterializedInternal<Object, Object, KeyValueStore<Bytes, byte[]>> materialized =
+            getDefaultMaterialized();
+        InternalTopologyBuilder tmpBuilder = new InternalTopologyBuilder();
+        tmpBuilder.addGlobalStore(
+            new TimestampedKeyValueStoreMaterializer<>(materialized).materialize().withLoggingDisabled(),
+            "sourceName",
+            null,
+            null,
+            null,
+            GLOBAL_STORE_TOPIC_NAME,
+            "processorName",
+            new KTableSource<>(GLOBAL_STORE_NAME, GLOBAL_STORE_NAME));
+        config = new StreamsConfig(getDefaultProperties());
+        globalStreamThread = new GlobalStreamThread(tmpBuilder.rewriteTopology(config).buildGlobalStateTopology(),
+            config,
+            tmpMockConsumer1,
+            new StateDirectory(config, time, true),
+            0,
+            new Metrics(),
+            new MockTime(),
+            "clientId",
+            stateRestoreListener);
+
+        initializeConsumer();
+        globalStreamThread.start();
+        globalStreamThread.shutdown();
+        long startTime = System.currentTimeMillis();
```

Comment:
@xin-au good idea with the timeouts, but I think we can simplify lines 310-329 down to a single `waitForCondition`.  Something along the lines of:
```java
TestUtils.waitForCondition(() -> !globalStreamThread.isAlive(),
            "Failed to shutdown a global stream thread,the global stream thread is still alive");
```